File: src/tests/ContainsGroupTest.java
package edu.sdsc.mmtf.spark.filters;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import java.util.Arrays;
import java.util.List;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.rcsb.mmtf.api.StructureDataInterface;

import edu.sdsc.mmtf.spark.io.MmtfReader;

public class ContainsGroupTest {
	private JavaSparkContext sc;
	private JavaPairRDD<String, StructureDataInterface> pdb;
	
	@Before
	public void setUp() throws Exception {
		SparkConf conf = new SparkConf().setMaster("local[*]").setAppName(ContainsGroupTest.class.getSimpleName());
	    sc = new JavaSparkContext(conf);
	    
	    // 1STP: only L-protein chain
	    // 1JLP: single L-protein chains with non-polymer capping group (NH2)
	    // 5X6H: L-protein and DNA chain
	    // 5L2G: DNA chain
	    // 2MK1: D-saccharide
	    List<String> pdbIds = Arrays.asList("1STP","1JLP","5X6H","5L2G","2MK1");
	    pdb = MmtfReader.downloadMmtfFiles(pdbIds, sc);
	}

	@After
	public void tearDown() throws Exception {
		sc.close();
	}

	@Test
	public void test1() {
	    pdb = pdb.filter(new ContainsGroup("BTN"));    
	    List<String> results = pdb.keys().collect();
	    
	    assertTrue(results.contains("1STP"));
	    assertFalse(results.contains("1JLP"));
	    assertFalse(results.contains("5X6H"));
	    assertFalse(results.contains("5L2G"));
	    assertFalse(results.contains("2MK1"));
	}
	
	@Test
	public void test2() {
	    pdb = pdb.filter(new ContainsGroup("HYP"));   
	    List<String> results = pdb.keys().collect();
	    
	    assertFalse(results.contains("1STP"));
	    assertTrue(results.contains("1JLP"));
	    assertFalse(results.contains("5X6H"));
	    assertFalse(results.contains("5L2G"));
	    assertFalse(results.contains("2MK1"));
	}
	
	@Test
	public void test3() {
	    pdb = pdb.filter(new ContainsGroup("HYP","NH2"));   
	    List<String> results = pdb.keys().collect();
	    
	    assertFalse(results.contains("1STP"));
	    assertTrue(results.contains("1JLP"));
	    assertFalse(results.contains("5X6H"));
	    assertFalse(results.contains("5L2G"));
	    assertFalse(results.contains("2MK1"));
	}
}

File: src/tests/containsGroupTest.py
#!/usr/bin/env python
'''
Simple example of reading an MMTF Hadoop Sequence file, filtering the entries \
by rWork,and counting the number of entries.

Authorship information:
__author__ = "Peter Rose"
__maintainer__ = "Mars Huang"
__email__ = "marshuang80@gmai.com:
__status__ = "Warning"
'''

import unittest
from pyspark import SparkConf, SparkContext
from src.main.MmtfReader import downloadMmtfFiles
from src.main.filters import containsGroup


path = '../full'

class testContainsDnaChainTest(unittest.TestCase):

    def setUp(self):
        conf = SparkConf().setMaster("local[*]").setAppName('testContainsAlternativeLocations')
        pdbIds = ['1STP','1JLP','5X6H','5L2G','2MK1']
        self.sc = SparkContext(conf=conf)
        self.pdb = downloadMmtfFiles(pdbIds,self.sc)


    def test1(self):
        pdb_1 = self.pdb.filter(containsGroup('BTN'))
        results_1 = pdb_1.keys().collect()

        self.assertTrue('1STP' in results_1)
        self.assertFalse('1JLP' in results_1)
        self.assertFalse('5X6H' in results_1)
        self.assertFalse('5L2G' in results_1)
        self.assertFalse('2MK1' in results_1)


    def test2(self):
        pdb_2 = self.pdb.filter(containsGroup('HYP'))
        results_2 = pdb_2.keys().collect()

        self.assertFalse("1STP" in results_2);
        self.assertTrue("1JLP" in results_2);
        self.assertFalse("5X6H" in results_2);
        self.assertFalse("5L2G" in results_2);
        self.assertFalse("2MK1" in results_2);


    def test3(self):
        pdb_3 = self.pdb.filter(containsGroup('HYP','NH2'))
        results_3 = pdb_3.keys().collect()

        self.assertFalse("1STP" in results_3);
        self.assertTrue("1JLP" in results_3);
        self.assertFalse("5X6H" in results_3);
        self.assertFalse("5L2G" in results_3);
        self.assertFalse("2MK1" in results_3);


    def tearDown(self):
        self.sc.stop()


if __name__ == '__main__':
    unittest.main()
